src/components/CommentsAdder.jsx

```diff
@@ -1,7 +1,86 @@
-import { useState } from "react";
+import { useState, useContext } from "react";
 import { useParams } from "react-router-dom";
-import { postComment } from "../utils/api";
+import { UserContext } from "../contexts/User";
+import { postCommentByArticleId } from "../utils/api";
+import '../styles/CommentAdder.css'
 
-const CommentAdder = ({setComments}) => {};
+const CommentAdder = ({ setComments, setCommentCount }) => {
+  const { article_id } = useParams();
+  const { user } = useContext(UserContext);
+  const [body, setBody] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+  const [err, setErr] = useState(null);
+  const [successMessage, setSuccessMessage] = useState("");
+
+  const handleBodyChange = (event) => {
+    setBody(event.target.value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    setIsLoading(true);
+    setErr(null);
+    setSuccessMessage("");
+
+    const newComment = {
+      body,
+      author: user,
+      article_id,
+      votes: 0,
+      created_at: new Date().toISOString(),
+    };
+    setComments((currentComments) => [newComment, ...currentComments]);
```

Comment:
This is a nice way to optimistically render the comment, but what does the post comment api endpoint give us back? Could this be used instead as this comes with a comment_id already?